bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/widgets/Tree.java

```diff
@@ -4319,6 +4319,80 @@ void checkSetDataInProcessBeforeRemoval() {
 	}
 }
 
+boolean initPixbufSize(Image image) {
+	if (pixbufSizeSet || image == null) {
+		return false;
+	}
+	int iWidth, iHeight;
+	if (DPIUtil.useCairoAutoScale()) {
+		iWidth = image.getBounds().width;
+		iHeight = image.getBounds().height;
+	} else {
+		iWidth = image.getBoundsInPixels().width;
+		iHeight = image.getBoundsInPixels().height;
+	}
+	if (iWidth <= 0 || iHeight <= 0) {
+		return false;
+	}
+	pixbufSizeSet = true;
+	pixbufHeight = iHeight;
+	pixbufWidth = iWidth;
+	/*
+	 * Feature in GTK: a Tree with the style SWT.VIRTUAL has fixed-height-mode
+	 * enabled. This will limit the size of any cells, including renderers. In order
+	 * to prevent images from disappearing/being cropped, we update row heights when
+	 * the first image is set. Fix for bug 480261.
+	 */
+	if ((style & SWT.VIRTUAL) != 0) {
+		resetFixedRowHeight();
+	}
+	return true;
+}
+
+private void resetFixedRowHeight() {
+	long columnList = GTK.gtk_tree_view_get_columns(handle);
+	if (columnList == 0) {
+		return;
+	}
+
+	// set fixed width and height for all GtkCellRendererPixbufs
+	for (var colItem = columnList; colItem != 0; colItem = OS.g_list_next(colItem)) {
+		long column = OS.g_list_data(colItem);
+		var cellList = GTK.gtk_cell_layout_get_cells(column);
+		for (var cellItem = cellList; cellItem != 0; cellItem = OS.g_list_next(cellItem)) {
+			var renderer = OS.g_list_data(cellItem);
+			if (GTK.GTK_IS_CELL_RENDERER_PIXBUF(renderer)) {
+				GTK.gtk_cell_renderer_set_fixed_size(renderer, pixbufWidth, pixbufHeight);
+			}
+		}
+		OS.g_list_free(cellList);
+	}
+	OS.g_list_free(columnList);
+
+	// Executed in asyncExec because when this method is invoked from checkData(),
+	// then the "row_changed" signal is blocked (but we need it unblocked to invalidate
+	// existing rows).
+	display.asyncExec(() -> {
+		if (!isDisposed() && modelHandle != 0) {
+			// reset computed 'fixed_height' to '-1'
+			OS.g_object_set(handle, OS.fixed_height_mode, false, 0);
+			OS.g_object_set(handle, OS.fixed_height_mode, true, 0);
+
+			// to update height of the existing rows we need to invalidate them in gtk
+			// we do that by invoking gtk_tree_view_row_changed on each of them
+			long iter = OS.g_malloc(GTK.GtkTreeIter_sizeof());
+			if (GTK.gtk_tree_model_get_iter_first(modelHandle, iter)) {
+				int[] value = new int[1];
```

Comment:
[minor] unnecessary reallocation

Reply:
Sorry, but I don't understand.  
Unnecessary reallocation of what?  
As I can see, both `iter` and `value` are allocated once.  
Could you clarify it, please.

Reply:
I had read `int[] value = new int[1];` as being a part of loop iteration. I was wrong. Sorry about that.

Reply:
no problem